### PR TITLE
Don't fire SRD() events until target peer identity has been verified.

### PR DESCRIPTION
**Pull request by jan-ivar**
Fixes https://github.com/w3c/webrtc-pc/issues/2173.

***
<a href="https://pr-preview.s3.amazonaws.com/jan-ivar/webrtc-pc/pull/2251.html" title="Last updated on Aug 3, 2019, 12:47 AM UTC (bdf95fd)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2251/f665b4d...jan-ivar:bdf95fd.html" title="Last updated on Aug 3, 2019, 12:47 AM UTC (bdf95fd)">Diff</a>

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -1417,6 +1417,17 @@ <h4>Set the RTCSessionDescription</h4>
...
+              <li>
+                <p>If <var>description</var> is set as a remote description and
+                is applied successfully, and a <a>target peer identity</a> is
+                set, then the user agent MUST wait for
+                <a href="#dfn-validate-the-identity">identity validation</a>
+                to complete before proceeding to the next step. If
+                <a href="#dfn-validate-the-identity">identity validation</a>
+                fails, <a>reject</a> <var>p</var> with a newly
+                <a data-link-for="exception" data-lt="create">created</a>
+                <code>OperationError</code>, and abort these steps.</p>
```
Comment (henbos) [inline webrtc.html]: If the description was applied successfully, but we reject the promise and abort the steps, won't that mean that the JavaScript layer and the JSEP layer are out-of-sync? I.e. there may be JSEP transceivers or m= sections or whatnot, but no JavaScript equivalent RTCRtpTransceivers?

Comment (henbos) [reply webrtc.html]: Do we need to tell JSEP to roll back?

File: webrtc.html [hunk 2]
```diff
@@ -1417,6 +1417,17 @@ <h4>Set the RTCSessionDescription</h4>
                   </li>
                 </ol>
               </li>
+              <li>
+                <p>If <var>description</var> is set as a remote description and
```
Comment (alvestrand) [inline webrtc.html]: It doesn't make sense to apply the description before you have verified the identity, I think.

Comment (jan-ivar) [reply webrtc.html]: @alvestrand Good point. @martinthomson WDYT?

File: webrtc.html (hunk 1 above)
Comment (jan-ivar) [reply webrtc.html]: You and @alvestrand make the same observation.

In practice if your peer connection has a target peerIdentity (cannot be changed), and you get a bad-identity remote description, you're most likely toast anyway and will want to discard the peer connection, unless you there's hope you'll get a valid offer reasonably soon thereafter, in which case you apply SRD again (which is legal).

A corner case may be a previously established connection without a target peerIdentity, then you set one, and then you get a bad-identity remote description. In this case I don't know whether having JSEP apply successfully is any worse than not applying it.

To be safe we should probably just hold off applying anything until the identity is validated. Right now Firefox does it in parallel with SRD, an optimization with little benefit, I suspect.

**Comment by henbos** [comment]
Start the process before applying the description

**Comment by alvestrand** [comment]
This is OBE based on the TPAC 2019 plan of moving the normative language into the identity spec.

**Pull request closed without merging**